torch/nn/modules/_functions.py

```diff
@@ -96,3 +96,47 @@ def backward(self, grad_output):
             grad_bias = None
 
         return grad_input, grad_weight, grad_bias, None, None, None, None, None, None
+
+    @staticmethod
+    def convert_sync_batchnorm(self, module, process_group=None):
+        r"""Helper function to convert `torch.nn.BatchNormND` layer in the model to
+        `torch.nn.SyncBatchNorm` layer.
+
+        Args:
+            module (nn.Module): containing module
+            process_group (optional): process group to scope synchronization,
+        default is the whole world
+
+        Returns:
+            The original module with the converted `torch.nn.SyncBatchNorm` layer
+
+        Example::
+
+            >>> # Network with nn.BatchNorm layer
+            >>> module = torch.nn.Sequential(
+            >>>            torch.nn.Linear(20, 100),
+            >>>            torch.nn.BatchNorm1d(100)
+            >>>          ).cuda()
+            >>> # creating process group (optional)
+            >>> # process_ids is a list of int identifying rank ids.
+            >>> process_group = torch.distributed.new_group(process_ids)
+            >>> sync_bn_module = convert_sync_batchnorm(module, process_group)
+
+        """
+        module_output = module
+        if isinstance(module, torch.nn.modules.batchnorm._BatchNorm):
+            module_output = torch.nn.SyncBatchNorm(module.num_features,
+                                                   module.eps, module.momentum,
+                                                   module.affine,
+                                                   module.track_running_stats,
+                                                   process_group)
+            if module.affine:
+                module_output.weight.data = module.weight.data.clone().detach()
+                module_output.bias.data = module.bias.data.clone().detach()
+            module_output.running_mean = module.running_mean
+            module_output.running_var = module.running_var
+            module_output.num_batches_tracked = module.num_batches_tracked
+        for name, child in module.named_children():
+            module_output.add_module(name, convert_sync_batchnorm(child))
```

Comment:
This recursive call needs to be fixed.

Reply:
True. Totally missed that. Also, to make sure that the relevant tests pass, I just need to use the relevant test files in the `tests` directory right?

Reply:
@iArunava yes, just run `python test_torch.py` etc and it should be fine

Reply:
I have made the changes required, also all the places here: https://github.com/pytorch/pytorch/search?q=convert_sync_batchnorm&unscoped_q=convert_sync_batchnorm 

`torch.nn.convert_sync_batch_norm` is replaced by `torch.nn.SyncBatchNorm.convert_sync_batch_norm` as its a `@classmethod` Is should be accessible in that way. Is that okay?

And before pushing I want to run the tests, but I am not able to run the tests, I keep getting cannot import module `istuple` from the `torch._six.py` file, I can't find the reason why, everything else from `_six.py` is imported okay. Can you please help @fmassa ? I am quite sure I am missing something here.